rs/bitcoin/validation/src/header.rs

```diff
@@ -283,22 +283,23 @@ fn compute_next_difficulty(
     // actual_interval will deviate slightly from 2 weeks. Our goal is to
     // readjust the difficulty target so that the expected time taken for the next
     // 2016 blocks is again 2 weeks.
-    let actual_interval = prev_header.time - last_adjustment_time;
+    let actual_interval = (prev_header.time as i64) - (last_adjustment_time as i64);
     let mut adjusted_interval = actual_interval;
 
     // The target_adjustment_interval_time is 2 weeks of time expressed in seconds
-    let target_adjustment_interval_time: u32 = DIFFICULTY_ADJUSTMENT_INTERVAL * TEN_MINUTES; //Number of seconds in 2 weeks
+    let target_adjustment_interval_time: i64 =
+        (DIFFICULTY_ADJUSTMENT_INTERVAL * TEN_MINUTES) as i64; //Number of seconds in 2 weeks
 
     // Adjusting the actual_interval to [0.5 week, 8 week] range in case the
     // actual_interval deviates too much from the expected 2 weeks.
-    adjusted_interval = u32::max(adjusted_interval, target_adjustment_interval_time / 4);
-    adjusted_interval = u32::min(adjusted_interval, target_adjustment_interval_time * 4);
+    adjusted_interval = i64::max(adjusted_interval, target_adjustment_interval_time / 4);
```

Comment:
NIT: use `clamp` instead of `max`+`min`.

```rust
let adjusted_interval = actual_interval
    .clamp(TARGET_ADJUSTMENT_INTERVAL / 4, TARGET_ADJUSTMENT_INTERVAL * 4);
```

Reply:
done thanks